File: app/src/main/java/com/example/android/popularmovies/adapters/ReviewsAdapter.java
package com.example.android.popularmovies.adapters;


import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.android.popularmovies.R;
import com.example.android.popularmovies.model.Review;
import com.example.android.popularmovies.model.ReviewItem;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ReviewsAdapter extends RecyclerView.Adapter<ReviewsAdapter.ReviewsAdapterViewHolder> {


    private Context context;
    private List<ReviewItem> mReview;

    public ReviewsAdapter() {
    }

    public ReviewsAdapter(Context context) {
        this.context = context;
    }

    @Override
    public ReviewsAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_reviews, parent, false);
        view.setFocusable(true);
        return new ReviewsAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ReviewsAdapterViewHolder holder, int position) {
        holder.tvReview.setText(mReview.get(position).getContent());
        holder.tvUsername.setText(mReview.get(position).getAuthor());
    }

    @Override
    public int getItemCount() {
        if(mReview != null)
            return mReview.size();
        return 0;
    }

    class ReviewsAdapterViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.tvUsername)
        TextView tvUsername;
        @BindView(R.id.tvReview)
        TextView tvReview;

        public ReviewsAdapterViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

    }

    public void setData(List<ReviewItem> review) {
        mReview= review;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/android/popularmovies/adapters/ReviewsAdapter.kt
package com.example.android.popularmovies.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.android.popularmovies.R
import com.example.android.popularmovies.model.Review
import com.example.android.popularmovies.model.ReviewItem

import java.util.ArrayList

import butterknife.BindView
import butterknife.ButterKnife
import kotlinx.android.synthetic.main.list_item_reviews.view.tvReview
import kotlinx.android.synthetic.main.list_item_reviews.view.tvUsername

class ReviewsAdapter : RecyclerView.Adapter<ReviewsAdapter.ReviewsAdapterViewHolder> {

  private var context: Context? = null
  private var mReview: List<ReviewItem>? = null

  constructor() {}

  constructor(context: Context?) {
    this.context = context
  }

  override fun onCreateViewHolder(
    parent: ViewGroup,
    viewType: Int
  ): ReviewsAdapterViewHolder {
    val view = LayoutInflater.from(parent.context)
        .inflate(R.layout.list_item_reviews, parent, false)
    view.isFocusable = true
    return ReviewsAdapterViewHolder(view)
  }

  override fun onBindViewHolder(
    holder: ReviewsAdapterViewHolder,
    position: Int
  ) {
    holder.tvReview!!.text = mReview!![position]
        .content
    holder.tvUsername!!.text = mReview!![position]
        .author
  }

  override fun getItemCount(): Int {
    return if (mReview != null) mReview!!.size else 0
  }

  inner class ReviewsAdapterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    val tvUsername = itemView.tvUsername;
    val tvReview = itemView.tvReview;

  }

  fun setData(review: List<ReviewItem>) {
    mReview = review
    notifyDataSetChanged()
  }
}
